### ISSUE TITLE
helm v3.6.1 and acr stopped working

### ISSUE THREAD

**Issue by josefschabasser** [Azure/azure-cli#18632]
Hello!

The latest version of helm (v3.6.1) changed how credentials are handled. Now `az acr helm repo add` generates non-working configuration. Please add support for helm flag `--pass-credentials` to set `pass_credentials_all: true` in `~/.config/helm/repositories.yaml`. Without this entry, `helm install/upgrade/pull` and similar fail with error 401/unauthorized.

for more information please see [this helm issue](https://github.com/helm/helm/issues/9857)

**Comment by yonzhan** [comment]
route to service team

**Comment by josefschabasser** [comment]
MS just reverted back to an image with helm v3.6.0. I think this change isn't needed, as the new helm version isn't used (for now), `az acr helm repo add` is deprecated and manually adding the repository is veeery easy.

Please see the followin snippet to get similar results:
```bash
REPO=MYREPO

CRED=($(az acr credential show --name ${REPO} --query '{username: username, password: passwords[0].value}' -o tsv))
HOST=$(az acr show --name ${REPO} --query 'loginServer' -o tsv)
REPO_URL="https://${HOST}/helm/v1/repo"
ARGS=()

if [[ $(semver -r '>=3.6.1' $(helm version --short)) ]]; then
  echo "helm v3.6.1 or newer detected, using '--pass-credentials'"
  ARGS+=('--pass-credentials')
fi

helm repo add ${REPO} ${REPO_URL} \
  --username ${CRED[0]} \
  --password ${CRED[1]} \
  ${ARGS[@]}
```

PRO:
* no use of deprecated commands
* credentials are valid until changed

CON:
* more than a single line
* credentials do not become invalid after 1 hour (security measure of azure-cli is gone)

**Comment by josefschabasser** [comment]
UPDATE: there was a bug which is fixed in helm v3.6.2. This issue can now be closed.
